Documentation/FrameworkOverview.md

```diff
@@ -8,65 +8,126 @@ learning about new modules and finding more specific documentation.
 For examples and help understanding how to use RAC, see the [README][] or
 the [Design Guidelines][].
 
-## Streams
-
-A **stream**, represented by the [RACStream][] abstract class, is any series of
-object values.
-
-Values may be available immediately or in the future, but must be retrieved
-sequentially. There is no way to retrieve the second value of a stream without
-evaluating or waiting for the first value.
-
-Streams are [monads][]. Among other things, this allows complex operations to be
-built on a few basic primitives (`-bind:` in particular). [RACStream][] also
-implements the equivalent of the [Monoid][] and [MonadZip][] typeclasses from
-[Haskell][].
-
-[RACStream][] isn't terribly useful on its own. Most streams are treated as
-[signals](#signals) or [sequences](#sequences) instead.
-
 ## Signals
 
-A **signal**, represented by the [RACSignal][] class, is a _push-driven_
-[stream](#streams).
+A **signal**, represented by the [Signal][] class, is any series of events 
+over time that can be observed.
+
+Signals are generally used to represent event streams that are already “in progress”,
+like notifications, user input, etc. As work is performed or data is received, 
+events are _sent_ on the signal, which pushes them out to any observers. 
+All observers see the events at the same time. 
 
-Signals generally represent data that will be delivered in the future. As work
-is performed or data is received, values are _sent_ on the signal, which pushes
-them out to any subscribers. Users must [subscribe](#subscription) to a signal
-in order to access its values.
+Users must [observe](#observers) a signal in order to access its events. 
+Observing a signal does not trigger any side effects. In other words, 
+signals are entirely producer-driven and push-based, and consumers (observers) 
+cannot have any effect on their lifetime. While observing a signal, the user 
+can only evaluate the events in the same order as they are sent on the signal -
+there is no random access to values of the stream.
 
-Signals send three different types of events to their subscribers:
+Signals send four different types of events to their observers:
 
- * The **next** event provides a new value from the stream. [RACStream][]
+ * The **next** event provides a new value from the stream. [Signal][]
    methods only operate on events of this type. Unlike Cocoa collections, it is
    completely valid for a signal to include `nil`.
  * The **error** event indicates that an error occurred before the signal could
-   finish. The event may include an `NSError` object that indicates what went
-   wrong. Errors must be handled specially – they are not included in the
-   stream's values.
+   finish. The event may include an signal specific `ErrorType` object that 
+   indicates what went wrong. Errors must be handled specially – they are not 
+   included in the stream's values.
```

Comment:
This should probably mention `NoError` as for signals and producers that can't error.